applications/MeshMovingApplication/tests/cpp_tests/test_mesh_velocity_calculation.cpp

```diff
@@ -0,0 +1,230 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:     BSD License
+//           Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+//
+
+#include <unordered_map>
+
+// Project includes
+#include "testing/testing.h"
+#include "includes/checks.h"
+#include "includes/model_part.h"
+#include "includes/kratos_parameters.h"
+#include "containers/model.h"
+#include "geometries/quadrilateral_2d_4.h"
+#include "processes/structured_mesh_generator_process.h"
+#include "utilities/variable_utils.h"
+#include "includes/mesh_moving_variables.h"
+
+// Application includes
+#include "custom_utilities/mesh_velocity_calculation.h"
+
+namespace Kratos {
+namespace Testing {
+
+namespace { // empty namespace to not pollute the testing-namespace with aux-functions
+
+typedef std::unordered_map<std::size_t, std::vector<double>> ResultsMapType;
+
+void CreateModelPartForMeshVelComputation(
+    Model& rModel,
+    const std::size_t BufferSize,
+    const double DeltaTime,
+    const bool RequiresMeshAcceleration)
+{
+    auto& r_model_part = rModel.CreateModelPart("MeshVelMP");
+
+    Node<3>::Pointer p_point_1 = Kratos::make_shared<Node<3>>(1, 0.0, 0.0, 0.0);
+    Node<3>::Pointer p_point_2 = Kratos::make_shared<Node<3>>(2, 0.0, 1.0, 0.0);
+    Node<3>::Pointer p_point_3 = Kratos::make_shared<Node<3>>(3, 1.0, 1.0, 0.0);
+    Node<3>::Pointer p_point_4 = Kratos::make_shared<Node<3>>(4, 1.0, 0.0, 0.0);
```

Comment:
Do you not need to add the nodes to the model part?

Reply:
nope
This is just the input for the StructuredMeshGenerator